File: app/src/main/java/com/example/project_note/DataBase/Note.java
package com.example.project_note.DataBase;


import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.io.Serializable;

@Entity(tableName = "note")
public class Note implements Serializable {

    @PrimaryKey(autoGenerate = true)
    private int id;
    private String title;
    private String detail;

    public Note( String title, String detail) {
        this.title = title;
        this.detail = detail;
    }

//    public Note(int id, String title, String detail) {
//        this.id = id;
//        this.title = title;
//        this.detail = detail;
//    }
    //    protected Note(Parcel in) {
//        id = in.readInt();
//        title = in.readString();
//        detail = in.readString();
//    }

//    public static final Creator<Note> CREATOR = new Creator<Note>() {
//        @Override
//        public Note createFromParcel(Parcel in) {
//            return new Note(in);
//        }
//
//        @Override
//        public Note[] newArray(int size) {
//            return new Note[size];
//        }
//    };

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDetail() {
        return detail;
    }

    public void setDetail(String detail) {
        this.detail = detail;
    }

//    @Override
//    public int describeContents() {
//        return 0;
//    }
//
//    @Override
//    public void writeToParcel(@NonNull Parcel dest, int flags) {
//        dest.writeString(title);
//        dest.writeString(detail);
//    }
}

File: app/src/main/java/com/example/project_note/DataBase/Note.kt
package com.example.project_note.DataBase

import androidx.room.ColumnInfo
import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Ignore
import androidx.room.PrimaryKey
import androidx.room.Relation
import androidx.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import java.io.Serializable

@Entity
data class Note(

    @ColumnInfo(name = "title")
    var title: String,
    @ColumnInfo(name = "detail")
    var detail: String,
    @ColumnInfo(name = "isSave")
    var isCheckImageSaveRD: Boolean = true,
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id_Note")
    var idNote : Int = 0) : Serializable{
}
@Entity
data class Image(

    var path: String,
    @ColumnInfo(name = "id_Note")
    var noteCreatorId : Int = -1,
    @ColumnInfo(name = "isSave")
    var isCheckImageSaveRD: Boolean = true,
    var pathFB: String = "null"
   ){
    @PrimaryKey(autoGenerate = true)
    var idImage: Int = 0
}



